controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1424,6 +1437,199 @@ func (r *BareMetalHostReconciler) manageHostPower(prov provisioner.Provisioner,
 	return actionUpdate{steadyStateResult}
 }
 
+// DataImage handler for attaching/detaching image.
+func (r *BareMetalHostReconciler) handleDataImageActions(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+	dataImage := &metal3api.DataImage{}
+	if err := r.Get(info.ctx, info.request.NamespacedName, dataImage); err != nil {
+		// DataImage does not exist or it may have been deleted
+		if k8serrors.IsNotFound(err) {
+			info.log.Info("dataImage not found")
+			return nil
+		}
+		// Error reading the object - requeue the request.
+		return actionError{fmt.Errorf("could not load dataImage, %w", err)}
+	}
+
+	// Set ControllerReference to DataImage
+	if !ownerReferenceExists(info.host, dataImage) {
+		if err := controllerutil.SetControllerReference(info.host, dataImage, r.Scheme()); err != nil {
+			return actionError{fmt.Errorf("could not set bmh as controller, %w", err)}
+		}
+		if err := r.Update(info.ctx, dataImage); err != nil {
+			return actionError{fmt.Errorf("failure creating dataImage resource, %w", err)}
+		}
+		// TODO(hroyrh) : Should we requeue at this point
```

Comment:
I think it should be simply `actionContinue{}` because we don't need a delay (we're ready to proceed immediately).

Reply:
Do we not need to wait for the `Update` to finish ?